examples/input/keyboard_input.rs

```diff
@@ -10,16 +10,16 @@ fn main() {
 }
 
 /// This system prints 'A' key state
-fn keyboard_input_system(keyboard_input: Res<Input<KeyCode>>) {
-    if keyboard_input.pressed(KeyCode::A) {
+fn keyboard_input_system(keyboard_input: Res<Input<Key>>) {
+    if keyboard_input.pressed(Key::Character("a".into())) {
```

Comment:
latest commit 54483f9389fb4c21c2632e13251d23e82fec1f68 breaks other examples, as it's quite a bit of changes to do, I'd like to have an agreement on the strategy to apply there, creating a `String` each time an input is even tested (so each frame, multiple times), sounds bad for performance.

there might be benchmarks about it I could try running 🤔 

Ideally, we would want to bring SmolStr in dependencies, and provide a FromReflect implementation for that. Would that be needed for this PR or could it be done in another ?

Reply:
current strategy is to implement Reflect* for SmolStr : https://github.com/bevyengine/bevy/pull/8771 ; until that SmolStr is merged, this winit PR is on hold.